experimental/python2.7/testdata/python27.yaml

```diff
@@ -23,588 +23,48 @@ commandTests:
-    exitCode: 0
-  - name: import_unittest
-    command: ["/usr/bin/python2.7", "-c", "import unittest"]
-    exitCode: 0
-  - name: import_urllib
-    command: ["/usr/bin/python2.7", "-c", "import urllib"]
-    exitCode: 0
-  - name: import_uu
-    command: ["/usr/bin/python2.7", "-c", "import uu"]
-    exitCode: 0
-  - name: import_uuid
-    command: ["/usr/bin/python2.7", "-c", "import uuid"]
-    exitCode: 0
-  - name: import_warnings
-    command: ["/usr/bin/python2.7", "-c", "import warnings"]
-    exitCode: 0
-  - name: import_wave
-    command: ["/usr/bin/python2.7", "-c", "import wave"]
-    exitCode: 0
-  - name: import_weakref
-    command: ["/usr/bin/python2.7", "-c", "import weakref"]
-    exitCode: 0
-  - name: import_webbrowser
-    command: ["/usr/bin/python2.7", "-c", "import webbrowser"]
-    exitCode: 0
-  - name: import_whichdb
-    command: ["/usr/bin/python2.7", "-c", "import whichdb"]
-    exitCode: 0
-  - name: import_wsgiref
-    command: ["/usr/bin/python2.7", "-c", "import wsgiref"]
-    exitCode: 0
-  - name: import_xdrlib
-    command: ["/usr/bin/python2.7", "-c", "import xdrlib"]
-    exitCode: 0
-  - name: import_xml
-    command: ["/usr/bin/python2.7", "-c", "import xml"]
-    exitCode: 0
-  - name: import_xmlrpclib
-    command: ["/usr/bin/python2.7", "-c", "import xmlrpclib"]
-    exitCode: 0
-  - name: import_zipfile
-    command: ["/usr/bin/python2.7", "-c", "import zipfile"]
-    exitCode: 0
-  - name: import_zipimport
-    command: ["/usr/bin/python2.7", "-c", "import zipimport"]
-    exitCode: 0
+  # import every module installed with the Python package
+  - name: import_everything
+    exitCode: 0
+    command:
+      - "/usr/bin/python2.7"
+      - "-c"
+      # multi-line YAML string with Python script that imports all modules that are installed.
+      # This ensures we have the right native library dependencies.
+      - |
+        import pkgutil
+
+        skip_modules = frozenset((
+          # Windows-specific modules
+          'ctypes.wintypes',
+          'distutils.command.bdist_msi',
+          'distutils.msvc9compiler',
+          'encodings.mbcs',
+
+          # depends on things not installed by default on Debian
+          'lib2to3.pgen2.conv',
+          '_LWPCookieJar',
+          '_MozillaCookieJar',
+        ))
+
+        def onerror(failed_module_name):
+          # ignore any errors from modules we are explicitly skipping (e.g. tkinter)
+          if failed_module_name in skip_modules:
+            return
+          raise Exception('failed to import module: {}'.format(repr(failed_module_name)))
+
+        for module_info in pkgutil.walk_packages(onerror=onerror):
+          # TODO: use module_info.name when Debian9 support is dropped
+          module_name = module_info[1]
+          # idlelib on Python3.5 depends on tkinter which is not installed by default
+          if module_name in skip_modules or module_name.endswith('.__main__'):
```

Comment:
If we are ignoring an import error in `def onerror` for `skip_modules`, why do we exclude `skip_modules` here? (I just would like to remove any unnecessary code.)

Reply:
Great suggestion! This is actually the "ignore modules" thing we need to keep. The other one, in the error handler, can go away. I think the difference may be that walk_packages needs to import *packages*, which I guess is what Python calls modules that contain other modules? walk_packages does not actually import the modules themselves.

I did remove the check, and I renamed and shuffled things around a bit, thanks to this suggestion.

Reply:
Ah, I spoke a bit too soon: On Python3, this error triggers for `tkinter`; on Python2.7, it does not. My intention was to make these tests as similar as possible. However, it turns out getting this to run with 4 different versions of Python is a bit tricky. I've simplified this for Python2, but Python3 still needs a bit of a gross hack.

Reply:
Thanks for the explanation. I think we don't have to make these as similar as possible. It seems like Python 2 is really dead by now, so I'd prefer having simpler code for Python 3, if that's possible.